src/org/thoughtcrime/securesms/connect/ForegroundDetector.java

```diff
@@ -35,9 +35,16 @@ public boolean isBackground() {
 
     @Override
     public void onActivityStarted(Activity activity) {
+        if (refs == 0) {
+            application.dcContext.maybeStartIo();
+            if (application.dcContext.isNetworkConnected()) {
+                new Thread(() -> {
+                    application.dcContext.maybeNetwork();
+                });
```

Comment:
```suggestion
                new Thread(() -> application.dcContext.maybeNetwork()).start();
```
You also need to start the thread

Reply:
good point!